ui/app/assets/javascripts/admin.js

```diff
@@ -0,0 +1,188 @@
+            data: data.migrations_completed
+        }],
+        credits: {
+            enabled: false
+        },
+        plotOptions: {
+            line: {
+                softThreshold: false
+            }
+        }
+    });
+}
+
+function updateFiledCompletedChart(weeks) {
+    $.ajax({
+        method: "GET",
+        url: "/admin/refresh_filed_chart",
+        data: {
+            weeks: weeks
+        },
+        success: function(result) {
+            drawFiledCompletedChart(result);
+        }
+    });
+}
+
+function drawApprovalTimeChart(data) {
+    $("#approval_time_chart").highcharts({
+        chart: {
+            type: "line"
+        },
+        title: {
+            text: null
+        },
+        xAxis: {
+            title: {
+                text: "Date"
+            },
+            categories: data.categories
+        },
+        yAxis: {
+            title: {
+                text: "Hours"
+            },
+            allowDecimals: false,
+            floor: 0,
+        },
+        tooltip: {
+            crosshairs: true,
+            valueSuffix: " hours"
+        },
+        series: [{
+            name: "Average Time",
+            data: data.average_approval_times,
+            connectNulls: true
+        }],
+        credits: {
+            enabled: false
+        }
+    });
+}
+
+function updateApprovalTimeChart(weeks) {
+    $.ajax({
+        method: "GET",
+        url: "/admin/refresh_approval_time_chart",
+        data: {
+            weeks: weeks
+        },
+        success: function(result) {
+            drawApprovalTimeChart(result);
+        }
+    });
+}
+
+function updateMetrics(cluster) {
+    $("#cluster_metrics_table").find("tbody").empty()
+    spinner.spin($("#spinner_target")[0])
+    $.ajax({
+        method: "GET",
+        url: "/admin/refresh_cluster_metrics",
+        data: {
+            cluster: cluster
+        },
+        success: function(result) {
+            var tbody = $("#cluster_metrics_table").find("tbody")
+            for (var db in result) {
+                var first = true;
+                for (var table in result[db]) {
+                    var data = result[db][table]
+                    var row = $("<tr>");
+                    if (first) {
+                        row.append($("<td>").text(db));
+                        first = false;
+                    } else {
+                        row.append($("<td>"));
+                    }
+                    row.append($("<td>").text(table))
+                    row.append($("<td>").text(data.times_altered))
+                    row.append($("<td>").text(toHHMMSS(data.average_alter_time)))
+                    row.append($("<td>").text(toHHMMSS(data.max_alter_time)))
+                    row.append($("<td>").text(toHHMMSS(data.min_alter_time)))
+
+                    tbody.append(row);
+                }
+            }
+            spinner.stop();
+        },
+        error: function(data) {
```

Comment:
let's append an error message to the html in this case. "Error loading data", centered in the table or something like that